stats/lib/handler.js

```diff
@@ -141,86 +143,18 @@ const notFound = (res) => {
  * @param {import('node:http').ServerResponse} res
- * @param {PgPools['api']} client
- * @param {string} minerId
+ * @param {string} SPARK_API_BASE_URL
  */
-const getRetrievableDealsForMiner = async (_req, res, client, minerId) => {
-  /** @type {{rows: {client_id: string; deal_count: number}[]}} */
-  const { rows } = await client.query(`
-    SELECT client_id, COUNT(cid)::INTEGER as deal_count FROM retrievable_deals
-    WHERE miner_id = $1 AND expires_at > now()
-    GROUP BY client_id
-    ORDER BY deal_count DESC, client_id ASC
-    `, [
-    minerId
-  ])
-
-  // Cache the response for 6 hours
-  res.setHeader('cache-control', `max-age=${6 * 3600}`)
-
-  const body = {
-    minerId,
-    dealCount: rows.reduce((sum, row) => sum + row.deal_count, 0),
-    clients:
-      rows.map(
-        // eslint-disable-next-line camelcase
-        ({ client_id, deal_count }) => ({ clientId: client_id, dealCount: deal_count })
-      )
-  }
-
-  json(res, body)
-}
-
-const getRetrievableDealsForClient = async (_req, res, client, clientId) => {
-  /** @type {{rows: {miner_id: string; deal_count: number}[]}} */
-  const { rows } = await client.query(`
-    SELECT miner_id, COUNT(cid)::INTEGER as deal_count FROM retrievable_deals
-    WHERE client_id = $1 AND expires_at > now()
-    GROUP BY miner_id
-    ORDER BY deal_count DESC, miner_id ASC
-    `, [
-    clientId
-  ])
-
+const redirectToSparkApi = async (req, res, SPARK_API_BASE_URL) => {
+  console.log('SPARK_API_BASE_URL', SPARK_API_BASE_URL)
+  console.log('req.url', req.url)
   // Cache the response for 6 hours
   res.setHeader('cache-control', `max-age=${6 * 3600}`)
 
-  const body = {
-    clientId,
-    dealCount: rows.reduce((sum, row) => sum + row.deal_count, 0),
-    providers: rows.map(
-      // eslint-disable-next-line camelcase
-      ({ miner_id, deal_count }) => ({ minerId: miner_id, dealCount: deal_count })
-    )
-  }
-  json(res, body)
-}
-
-const getRetrievableDealsForAllocator = async (_req, res, client, allocatorId) => {
-  /** @type {{rows: {client_id: string; deal_count: number}[]}} */
-  const { rows } = await client.query(`
-    SELECT ac.client_id, COUNT(cid)::INTEGER as deal_count
-    FROM allocator_clients ac
-    LEFT JOIN retrievable_deals rd ON ac.client_id = rd.client_id
-    WHERE ac.allocator_id = $1 AND expires_at > now()
-    GROUP BY ac.client_id
-    ORDER BY deal_count DESC, ac.client_id ASC
-    `, [
-    allocatorId
-  ])
-
-  // Cache the response for 6 hours
-  res.setHeader('cache-control', `max-age=${6 * 3600}`)
-
-  const body = {
-    allocatorId,
-    dealCount: rows.reduce((sum, row) => sum + row.deal_count, 0),
-    clients: rows.map(
-      // eslint-disable-next-line camelcase
-      ({ client_id, deal_count }) => ({ clientId: client_id, dealCount: deal_count })
-    )
-  }
-  json(res, body)
+  const location = new URL(req.url, SPARK_API_BASE_URL).toString()
+  res.setHeader('location', location)
+  res.statusCode = 302
+  res.end(location)
```

Comment:
```suggestion
  redirect(req, res, location, 302)
```

https://github.com/juliangruber/http-responders/blob/5820864da9cb4728d4d86e76184eb554a92c16ed/index.js#L15